src/backend/distributed/executor/multi_utility.c

```diff
@@ -1661,23 +1687,55 @@ ProcessVacuumStmt(VacuumStmt *vacuumStmt, const char *vacuumCommand)
  * statement needs distributed execution but contains unsupported options.
  */
 static bool
-IsSupportedDistributedVacuumStmt(Oid relationId, VacuumStmt *vacuumStmt)
+IsSupportedDistributedVacuumStmt(VacuumStmt *vacuumStmt, List *vacuumRelationIdList)
 {
 	const char *stmtName = (vacuumStmt->options & VACOPT_VACUUM) ? "VACUUM" : "ANALYZE";
 	bool distributeStmt = false;
+	ListCell *relationIdCell = NULL;
+	int distributedRelationCount = 0;
+	int vacuumedRelationCount = 0;
 
-	if (vacuumStmt->relation == NULL)
+	if (vacuumRelationIdList == NIL)
 	{
 		/* WARN for unqualified VACUUM commands */
 		ereport(WARNING, (errmsg("not propagating %s command to worker nodes", stmtName),
 						  errhint("Provide a specific table in order to %s "
 								  "distributed tables.", stmtName)));
+		return false;
 	}
-	else if (!OidIsValid(relationId) || !IsDistributedTable(relationId))
+
+	vacuumedRelationCount = list_length(vacuumRelationIdList);
+
+	foreach(relationIdCell, vacuumRelationIdList)
 	{
-		/* Nothing to do here; relation no longer exists or is not distributed */
+		Oid relationId = lfirst_oid(relationIdCell);
+		if (OidIsValid(relationId) && IsDistributedTable(relationId))
+		{
+			distributedRelationCount++;
+		}
 	}
-	else if (!EnableDDLPropagation)
+
+	/*
+	 * There can only be one distributed relation in the vacuum statement.
```

Comment:
I'd rephrase this as "we currently only support a single distributed relation in the `VACUUM` statement, because there _can_ be more than one, we just don't support it.